pkg/algorithm/nodeselect/v1alpha1/select_node.go

```diff
@@ -186,6 +159,46 @@ func (ac *Config) selectNode(nodeBlockDeviceMap map[string]*blockDeviceList) *no
 	// minRequiredDiskCount will hold the required number of disk that should be selected from a qualified
 	// node for specific pool type
 	minRequiredDiskCount := spcv1alpha1.DefaultDiskCount[ac.poolType()]
+
+	if ProvisioningType(ac.Spc) == ProvisioningTypeAuto {
+		// List all blockdeviceclaims related to spc
+		// NOTE: This list contains blockdevices on which csps' are already
+		// created filter and use it
+		bdcList, err := bdc.NewKubeClient().
+			WithNamespace(ac.Namespace).
+			List(metav1.ListOptions{LabelSelector: string(apis.StoragePoolClaimCPK) + "=" + ac.Spc.Name})
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to select node and blockdevices")
+		}
+		// customBDCList is a wrapper over blockdeviceclaim list
+		customBDCList := bdc.BlockDeviceClaimList{
+			ObjectList: bdcList,
+		}
+		// Form node and blockdevice topology for blockdevices which has bdc
+		claimedDevicesOnNode := customBDCList.GetBlockDeviceNamesByNode()
+		for claimedNodeName, bdNameList := range claimedDevicesOnNode {
+			qualifiedBDList := []string{}
```

Comment:
why we need this variable? can't we directly use bdNameList

Reply:
Yes we can reuse it.

Reply:
I will update it

Reply:
Basically it is useful only in partial bdc creation case instead of modifying `bdNameList`. Anyway I will update it.